.github/workflows/release_pr.yml

```diff
@@ -0,0 +1,43 @@
+name: Prepare Next Release
+on:
+  workflow_dispatch:
+    inputs:
+      release_tag:
+        description: 'Release Tag'
+        required: true
+        type: string
+env:
+  NEW_VERSION: ${{ github.event.inputs.release_tag }}
+jobs:
+  release_pr:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: main
+          fetch-depth: 0
+      - name: Modify for next release
+        run: |
+          chmod +x release.sh
+          ./release.sh ${{ env.NEW_VERSION }}
+          git diff
+      - name: Create PR
```

Comment:
why not directly pushing it to the mainline branch?

Reply:
The main branch seems can’t add exception for github action bot. The reference here is https://github.com/aws-amplify/amplify-android/pull/2554

Reply:
Resolve the problem for using PAT when push to protected branch. changed to directly pushing to main branch, tested this [action](https://github.com/awslabs/clickstream-android/actions/runs/5923706634/job/16059860806) and passed.